internal/controller/install/common_helpers.go

```diff
@@ -619,6 +619,32 @@ func isNil(i any) bool {
 	}
 }
 
+// deleteObjectIfNeeded will delete the object if it exists.
+func deleteObjectIfNeeded(
+	ctx context.Context,
+	client client.Client,
+	object client.Object,
+	componentName string,
+	logger logr.Logger,
+) error {
+	if isNil(object) {
+		return nil
+	}
+
+	err := client.Delete(ctx, object)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
```

Comment:
This whole `if/else` block can be replaced with the `return client.IgnoreNotFound(err)`

Reply:
Sounds good 4238007b3559cf8290d01ebb79a96acec350596f